boa/src/builtins/value/mod.rs

```diff
@@ -558,20 +554,16 @@ impl ValueData {
     /// Returns a copy of the Property.
     pub fn get_internal_slot(&self, field: &str) -> Value {
         let _timer = BoaProfiler::global().start_event("Value::get_internal_slot", "value");
-        let obj: Object = match *self {
-            Self::Object(ref obj) => {
-                let hash = obj.clone();
-                hash.into_inner()
-            }
-            Self::Symbol(ref obj) => {
-                let hash = obj.clone();
-                hash.into_inner()
-            }
-            _ => return Value::undefined(),
-        };
 
-        match obj.internal_slots.get(field) {
-            Some(val) => val.clone(),
+        let property = self
+            .as_object()
+            .and_then(|x| match x.internal_slots().get(field) {
+                Some(value) => Some(value.clone()),
+                None => None,
+            });
+
+        match property {
+            Some(value) => value,
             None => Value::undefined(),
         }
```

Comment:
I think this is more clear:
```suggestion
        property.unwrap_or_else(Value::undefined)
```

Reply:
In this case it's much cleaner :)